client/dashboard/sites/settings-repositories/connect-repository-form.tsx

```diff
@@ -145,15 +159,24 @@ const RepositorySelector = ( {
 
 type GithubAccountSelectorProps = DataFormControlProps< ConnectRepositoryFormData > & {
 	onAddGithubAccount: () => void;
+	shouldRestoreFocus: boolean;
 };
 
 const GithubAccountSelector = ( {
 	field,
 	onChange,
 	data,
 	onAddGithubAccount,
+	shouldRestoreFocus,
 }: GithubAccountSelectorProps ) => {
 	const { id, getValue } = field;
+	const selectRef = useRef< HTMLSelectElement | null >( null );
+
+	useEffect( () => {
+		if ( selectRef.current && shouldRestoreFocus ) {
+			selectRef.current.focus();
+		}
+	}, [ selectRef, shouldRestoreFocus ] );
```

Comment:
I think we could omit `selectRef` here.

```suggestion
	}, [ shouldRestoreFocus ] );
```

Reply:
That's right, well spotted! Removing it

Reply:
[2163953](https://github.com/Automattic/wp-calypso/pull/106521/commits/21639533f65cca12ad0c175056ae85b897a37fa3)